frontend/simple/js/state.js

```diff
@@ -134,7 +189,9 @@ const actions = {
         return console.error(`bad action ${type} on ${contractType} (${contractId}):`, entry)
       }
       // TODO: verify each entry is signed by a group member
-      await db.addLogEntry(contractId, entry)
+      let hash = await db.addLogEntry(contractId, entry)
+      // If this is a duplicate the hash will be bogus and we should exit
+      if (!hash) { return }
```

Comment:
'bogus'? Is 'null' really that bogus?

Also, can that be written as `!hash && return` to get rid of the braces?